spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/store/channel/JacksonChannelMessageStorePreparedStatementSetter.java

```diff
@@ -0,0 +1,133 @@
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.jdbc.store.channel;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Types;
+
+import tools.jackson.core.JacksonException;
+import tools.jackson.databind.ObjectMapper;
+
+import org.springframework.integration.support.json.JacksonMessagingUtils;
+import org.springframework.messaging.Message;
+import org.springframework.util.Assert;
+
+/**
+ * A {@link ChannelMessageStorePreparedStatementSetter} implementation that uses Jackson
+ * to serialize {@link Message} objects to JSON format instead of Java serialization.
+ * <p>
+ * This implementation stores the entire message (including headers and payload) as JSON,
+ * with type information embedded using Jackson's {@code @class} property.
+ * <p>
+ * <b>IMPORTANT:</b> JSON serialization exposes message content in text format in the database.
+ * Ensure proper database access controls and encryption for sensitive data.
+ * Consider the security implications before using this in production with sensitive information.
+ * <p>
+ * The {@link ObjectMapper} is configured using {@link JacksonMessagingUtils#messagingAwareMapper(String...)}
+ * which includes custom serializers/deserializers for Spring Integration message types
+ * and embeds class type information for secure deserialization.
+ * <p>
+ * <b>Database Requirements:</b>
+ * This implementation requires modifying the MESSAGE_CONTENT column to a text-based type:
+ * <ul>
+ *     <li>PostgreSQL: Change from {@code BYTEA} to {@code JSONB}</li>
+ *     <li>MySQL: Change from {@code BLOB} to {@code JSON}</li>
+ *     <li>H2: Change from {@code LONGVARBINARY} to {@code CLOB}</li>
+ * </ul>
+ * See the reference documentation for schema migration instructions.
+ * <p>
+ * <b>Usage Example:</b>
+ * <pre>{@code
+ * &#64;Bean
+ * JdbcChannelMessageStore messageStore(DataSource dataSource) {
+ *     JdbcChannelMessageStore store = new JdbcChannelMessageStore(dataSource);
+ *     store.setChannelMessageStoreQueryProvider(new PostgresChannelMessageStoreQueryProvider());
+ *
+ *     // Enable JSON serialization (requires schema modification)
+ *     store.setPreparedStatementSetter(
+ *         new JacksonChannelMessageStorePreparedStatementSetter());
+ *     store.setMessageRowMapper(
+ *         new JacksonMessageRowMapper("com.example"));
+ *
+ *     return store;
+ * }
+ * }</pre>
+ *
+ * @author Yoobin Yoon
+ *
+ * @since 7.0
+ */
+public class JacksonChannelMessageStorePreparedStatementSetter extends ChannelMessageStorePreparedStatementSetter {
+
+	private final ObjectMapper objectMapper;
```

Comment:
Can we go our `JsonObjectMapper` API instead?
No one said that we are always going to work just with Jackson.